### PR TITLE
vendor: update git url handling with changes to buildkit

### PR DESCRIPTION
**Pull request by jsternberg**
The git ssh url parsing API changed a bit and broke how buildx utilized
the API. This updates the affected method with the new method of
identifying an SSH url.

### CODE REVIEW

**Review by jedevc — APPROVED** [review]

**Comment by jedevc** [comment]
The SSH stuff looks fine, but there is quite a large difference in other buildkit changes - maybe might be worth holding off on merging till after the next buildx release?

No strong feelings though, LGTM

**Comment by tonistiigi** [comment]
@jedevc Otoh we already have vendored master so would need to ship with an unreleased version anyway.

File: go.mod [hunk 1]
```diff
@@ -23,7 +23,7 @@ require (
 	github.com/google/uuid v1.3.0
 	github.com/hashicorp/go-cty-funcs v0.0.0-20200930094925-2721b1e36840
 	github.com/hashicorp/hcl/v2 v2.8.2
-	github.com/moby/buildkit v0.12.1-0.20230804094609-b49a8873179b
+	github.com/moby/buildkit v0.12.1-0.20230822160532-fdacdb78e93a
```
Comment (crazy-max) [inline go.mod:26]: https://github.com/docker/buildx/actions/runs/5979375406/job/16223405923?pr=2023#step:6:229

```
2023/08/25 18:52:10 buildkitd: flag provided but not defined: -otel-socket-path
2023/08/25 18:52:10 > stopped 2023-08-25 18:52:10.334278273 +0000 UTC m=+5.977077113 exit status 1 1
```

Related to https://github.com/moby/buildkit/pull/4101/commits/06d51f9b40e4586a2d176008d8d017fc1f3d5848

As we now vendor buildkit https://github.com/moby/buildkit/commit/fdacdb78e93af76b0b4e53bfe6a1514fba34a378 with this change we need to update https://github.com/docker/buildx/blob/da6662975f0de310887ba2f2c748cbd21753e871/Dockerfile#L9.

Probably using this digest as version matching the very same commit: https://github.com/moby/buildkit/actions/runs/5941283562/job/16115090302#step:7:1179

```
master@sha256:f860a4917188cc9918319dc058a32c968c087ecbd3acda9c4447c650eefa11df
```

Comment (crazy-max) [reply go.mod:26]: @jsternberg Can you to the vendor update in a separate commit with the Dockerfile update as well?

**Comment by jsternberg** [comment]
This and some other changes are going to be part of https://github.com/docker/buildx/pull/2035 so I'm going to close this one and allow that PR to take this one's place.

**Pull request closed without merging**